### PR TITLE
[JENKINS-33596] disable UDP broadcast & jmDNS by default, make configurable

### PR DESCRIPTION
**Pull request by kzantow**
Moves configuration for jmDNS and UDP broadcast from `Jenkins` constructor to global security configuration items.

Addresses: https://issues.jenkins-ci.org/browse/JENKINS-33596

TODO:
- [ ] Ensure upgrading from prior versions results in these services being enabled as they were

### CODE REVIEW

**Comment by daniel-beck** [comment]
Would it make sense to introduce a `network-discovery` plugin, that bundles these features and possibly a fixed version https://github.com/jenkinsci/jenkins/pull/1301?

CC @jglick who IIRC said in the past that UDP broadcast and DNS multicast should be a new plugin.

**Comment by daniel-beck** [comment]
FWIW I think even if it's a separate plugin, having UI options for this is useful. So these changes (detach + configurable) are complementary.

**Comment by kzantow** [comment]
@daniel-beck right, I was taking a piecemeal approach to this... side note: I don't want to be the maintainer of the network-discovery plugin :)

File: core/src/main/java/hudson/UDPBroadcastThread.java [hunk 1]
```diff
@@ -144,4 +203,70 @@ public void shutdown() {
...
+        public void setUdpBroadcastPort(int udpBroadcastPort) {
+            this.udpBroadcastPort = udpBroadcastPort;
+        }
+        
+        @Override
+        public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
+            Jenkins jenkins = Jenkins.getInstance();
+            jenkins.checkPermission(Jenkins.ADMINISTER);
+
+            // for compatibility reasons, the actual value is stored in Jenkins
```
Comment (amuniz) [inline core/src/main/java/hudson/UDPBroadcastThread.java]: But the fields were removed from Jenkins, so I guess this information is stored now in its own XML file, no?

**Pull request closed without merging**